app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/CreateRecordController.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.controllers
+
+import cats.data.EitherT
+import cats.implicits._
+import com.google.inject.Inject
+import play.api.Logging
+import play.api.libs.json.Json.toJson
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents, Request, Result}
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.request.CreateRecordRequest
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.errors.ErrorResponse
+import uk.gov.hmrc.tradergoodsprofilesrouter.service.{RouterService, UuidService}
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.{ApplicationConstants, HeaderNames, ValidationSupport}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CreateRecordController @Inject() (
+  cc: ControllerComponents,
+  routerService: RouterService,
+  uuidService: UuidService
+)(implicit
+  executionContext: ExecutionContext
+) extends BackendController(cc)
+    with Logging {
+
+  def create: Action[JsValue] = Action.async(parse.json) { implicit request =>
+    val result = for {
+      _ <- validateClientId
+
+      createRecordRequest <- validateRequestBody(request)
```

Comment:
I think we need to handle it here too because the frontend because based on [this architecture diagram](https://confluence.tools.tax.service.gov.uk/display/TGP/TGP+Digital+Services+-+High+Level+Design), the frontend calls the router directly. 
@AmarkumarJagtap @taylors-version what do you think?

Reply:
Ok, I see your point Paul, so probably the external API should not validate the payload, as otherwise we will do it in two different place and just repeat it